src/main/scala/io/iohk/ethereum/blockchain/sync/fast/FastSync.scala

```diff
@@ -567,16 +567,13 @@ class FastSync(
     }
 
     // TODO [ETCM-676]: Move to blockchain and make sure it's atomic
-    private def discardLastBlocks(startBlock: BigInt, blocksToDiscard: Int): Unit = {
+    private def discardLastBlocks(startBlock: BigInt, blocksToDiscard: Int): Unit =
       // TODO (maybe ETCM-77): Manage last checkpoint number too
-      appStateStorage.putBestBlockNumber((startBlock - blocksToDiscard - 1).max(0)).commit()
```

Comment:
why is saving to the storage removed here?🤔

Reply:
After I did all the changes I had a bunch of tests failing in FastSyncItSpec, because the best blocknumber was always a bit inferior than expected, and the issue was that line.
Now that we don't have the caches anymore and always save to storage that instructions was bringing issues.